julia.html.markdown

```diff
@@ -145,171 +145,394 @@ SomeOtherVar123! = 6 # => 6
 catch e
     println(e)
 end
 
 # Use the get method to avoid that error by providing a default value
 # get(dictionary,key,default_value)
-get(filled_dict,"one",4) # => 1
-get(filled_dict,"four",4) # => 4
+get(filled_dict, "one", 4) # => 1
+get(filled_dict, "four", 4) # => 4
 
 # Use Sets to represent collections of unordered, unique values
-empty_set = Set() # => Set{Any}()
+empty_set = Set() # => Set{Any}[]
 # Initialize a set with values
-filled_set = Set([1,2,2,3,4]) # => Set{Int64}(1,2,3,4)
+filled_set = Set([1,2,2,3,4]) # => Set([4, 2, 3, 1])
 
 # Add more values to a set
-push!(filled_set,5) # => Set{Int64}(5,4,2,3,1)
+push!(filled_set, 5) # => Set{Int64}([4,2,3,5,1])
+filled_set # => Set{Int64}([4,2,3,5,1])
 
 # Check if the values are in the set
-in(2, filled_set) # => true
-in(10, filled_set) # => false
+2 in filled_set # => true
+10 in filled_set # => false
 
 # There are functions for set intersection, union, and difference.
-other_set = Set([3, 4, 5, 6]) # => Set{Int64}(6,4,5,3)
-intersect(filled_set, other_set) # => Set{Int64}(3,4,5)
-union(filled_set, other_set) # => Set{Int64}(1,2,3,4,5,6)
-setdiff(Set([1,2,3,4]),Set([2,3,5])) # => Set{Int64}(1,4)
+other_set = Set([3, 4, 5, 6]) # => Set{Int64}([4,3,5,6])
+intersect(filled_set, other_set) # => Set{Int64}([4,3,5])
+filled_set ∩ other_set # => Set{Int64}([4,3,5]) # The same with ∩ (\cap)
+union(filled_set, other_set) # => Set{Int64}([4,2,3,5,6,1])
+filled_set ∪ other_set # => Set{Int64}([4,2,3,5,6,1]) # The same ∪ (\cup)
+setdiff(Set([1,2,3,4]), Set([2,3,5])) # => Set{Int64}([4,1])
+
+# Set operations work on arrays, too
+[1,2,3,4] ∩ [2,3,4] # => 3-element Array{Int64,1}: [2,3,4]
+
+
+
+
+###########
+## Strings
+###########
+
+# Strings are created with "
+s = "This is a string."
+
+"""You can
+have multiline strings, too."""
+
+# Character literals are written with '
+'a'
+
+# Strings can be indexed like an array of characters
+s[1] # => 'T', a Char
+s[1:4] # => "This", a String
+
+# Indexing a unicode string in a wrong place will throw an arror,
+# rather than returning gibberish
+utf8string = "Θεσσαλονίκη"
+utf8string[1] # => 'Θ'
+try
+    utf8string[2] # => UnicodeError: invalid character index
+catch e
+    println(e)
+end
+utf8string[3] # => 'ε'
+
+# Iterating over Strings works as expected, even for Strings with
+# unicode characters
+[ c for c in utf8string ] # => 11-element Array{Char,1}:
+                          # ['Θ','ε','σ','σ','α','λ','ο','ν','ί','κ','η']
+
+
+# Strings are immutable
+try
+    s[1] = "T" # => MethodError: no method matching
+               #    setindex!(::String, ::String, ::Int64)
+catch e
+    println(e)
+end
+
+
+# $ can be used for string interpolation:
+"pi is $pi" # => "pi is π = 3.1415926535897..."
+"2 + 2 = $(2 + 2)" # => "2 + 2 = 4"
+# You can put any Julia expression inside the parentheses.
+
+# Another way to format strings is the printf macro.
+@printf "%d is less than %f" 4.5 5.3 # prints "5 is less than 5.300000"
```

Comment:
Using %d here was probably the original error, and raises all sorts of issues on rounding, etc. If you change it to %g the the original comment is correct, and we see that %g works the same as it does in most other languages. I wold recommend making that correction rather than changing the 4.5 to 5 in the output comment. In other words:

@printf "%g is less than %f" 4.5 5.3 # prints "4.5 is less than 5.300000"

Reply:
Same issue in next chunk too, obviously...